### DISCUSSION TITLE
Auxvariable as a function of position along a specified distance/length in the model

### DISCUSSION THREAD

**Discussion by batodon** [idaholab/moose#23333] [Q&A Modules: General] [answered]
Hello, I’m using the `ParsedFunction` to implement an `Auxvariable` that depends on a specific execution time (e.g., `timestep_end`) and position along a distance (e.g., along the x-axis) in the model. My function should look like this:
`Z= R/sqrt(t)`, where `R` is the position along the specified length and `t` is the execution time. Could anyone please help with possible leads? Here is a snippet of my code. Thank you.

```
[Functions]
  [sol_variable]
   type = ParsedFunction 
   value = ??
  []
[] 

[AuxKernels]
 [Z]
    type = FunctionAux
    variable = Z
    function = sol_variable
 []
[]

[AuxVariables]
 [Z]
    order = FIRST
    family = LAGRANGE
  []
[]
```

**Comment by GiudGiud** [comment, marked as answer]
timestep_end is not a variable

It should be t for time

**Reply by batodon** [reply]
Sure @GiudGiud. Sorry, that value in the `ParsedFunction` is not correct. It is just an 'example' to show what I really want to implement 😊.

**Reply by GiudGiud** [reply]
do this then (with a new version of MOOSE, we changed the parameters)
```
[Functions]
  [sol_variable]
   type = ParsedFunction 
   expression = R/sqrt(t)
   symbol_names = R
   symbol_values = "${R}"
  []
[] 
```
R being defined somewhere else in the input file as
```
R = some_value
```

if R is not a constant. you ll need to adapt this

**Reply by batodon** [reply]
Ok. I got you. How will I define `R`, though? `R` is the position along the entire length of my domain (for example, the x-axis). I want to output my final result at (let's say) 10 positions/points along the x-axis.

**Reply by GiudGiud** [reply]
oh then use x instead of R

you can then get rid of these two lines
   symbol_names = R
   symbol_values = "${R}"

for output, use a vectorpostprocessor
https://mooseframework.inl.gov/source/vectorpostprocessors/LineFunctionSampler.html

**Reply by batodon** [reply]
Great, this is exactly what I want (the vectorpostprocessor). Thanks, Guillaume.  👍

**Reply by batodon** [reply]
Hello @GiudGiud, probably trivial, but I tried using the log function in my `ParsedFunction`, but the input file did not accept it. How could I do this, please? I want a function like `Z= log(x/sqrt(t))`. Thank you.

**Reply by GiudGiud** [reply]
It should
see this manual for what is supported
http://warp.povusers.org/FunctionParser/fparser.html
